Game2048Base/src/test/java/models/grid2048/Grid2048Test.java

```diff
@@ -0,0 +1,109 @@
+package models.grid2048;
+
+import javalib.worldimages.Posn;
+import models.square.*;
+import org.junit.Test;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class Grid2048Test {
+    EmptyTile tEmpty;
+    Tile t2, t4, t8, t16;
+    Grid2048 g0, g1, g2;
+    Square[] rowEmpty0, rowEmpty1, rowEmpty2, rowEmpty3,
+            row0, row1, row2, row3;
+
+    void initData() {
+        tEmpty = new EmptyTile();
+        t2 = new Tile(2);
+        t4 = new Tile(4);
+        t8 = new Tile(8);
+        t16 = new Tile(16);
+
+        rowEmpty0 = new Square[]{
+                new EmptyTile(new Posn(0,0)),
+                new EmptyTile(new Posn(0,1)),
+                new EmptyTile(new Posn(0,2)),
+                new EmptyTile(new Posn(0,3))};
+        rowEmpty1 = new Square[]{
+                new EmptyTile(new Posn(1,0)),
+                new EmptyTile(new Posn(1,1)),
+                new EmptyTile(new Posn(1,2)),
+                new EmptyTile(new Posn(1,3))};
+        rowEmpty2 = new Square[]{
+                new EmptyTile(new Posn(2,0)),
+                new EmptyTile(new Posn(2,1)),
+                new EmptyTile(new Posn(2,2)),
+                new EmptyTile(new Posn(2,3))};
+        rowEmpty3 = new Square[]{
+                new EmptyTile(new Posn(3,0)),
+                new EmptyTile(new Posn(3,1)),
+                new EmptyTile(new Posn(3,2)),
+                new EmptyTile(new Posn(3,3))};
+        row0 = new Square[] {t2, tEmpty, tEmpty, tEmpty};
+        row1 = new Square[] {t2, t4, tEmpty, t16};
+        row2 = new Square[] {tEmpty, tEmpty, t8, t2};
+        row3 = new Square[] {t4, tEmpty, tEmpty, t16};
+
+        g0 = new Grid2048();
+        g1 = new Grid2048(new Square[][] {rowEmpty0, rowEmpty1, rowEmpty2, rowEmpty3});
+        g2 = new Grid2048(new Square[][] {row0, row1, row2, row3});
+    }
+
+    @Test
+    public void testRandomnessOfTilesToGrid() {
```

Comment:
This test name doesn't seem to correlate with what's under test. It looks like you're testing an error that occurs if there are too many tiles.

One common practice is to preface unit test names with 'it', where 'it' refers to the class under test, in this case Grid2048. At the end of the day, you can choose whatever naming convention is most intuitive to you, but consistency is key. 

In this case, maybe `itDetectsAndThrowsErrorOnTooManyTiles`